UnitedIncome/components
# Input 'validate' Prop

### Description
> Please provide a brief description of your changes below.

Created a new `validate` prop for the `Input` component. The idea here is that a user could pass in a simple function that returns either an empty string (valid input) or an error message and this would automatically grant them a few things, namely:

- Input validation based on the error message
- Error message automatically being set to the message of the `validate` func

All while not breaking existing functionality.

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

Download the branch and play with the new `Input` story labeled `validate`

### Comments
> Any additional comments that will aid with the review process.

For reference, this is what is passed to the `validate` story:

```javascript
validate={(input) => {
    if (input.length < 4) {
        return '';
    } else {
        return 'Please enter a valid small string';
    }
}}
```

I realize that the `maxLength` prop could accomplish the above, this was just meant to be a quick example.
